hr_holidays_leave_auto_approve/models/hr_leave.py

```diff
@@ -12,6 +12,13 @@ def _check_approval_update(self, state):
             return
         return super()._check_approval_update(state)
 
+    @api.multi
+    def apply_auto_approve_policy(self):
+        for record in self:
+            policy = record.holiday_status_id.auto_approve_policy
+            if (record.can_approve and policy == 'hr') or policy == 'all':
```

Comment:
Reuse `_should_auto_approve `? 
```
if record._should_auto_approve():
    record.sudo().action_approve()
```

Reply:
The whole function could be done in one line like this
```python
self.filtered(lambda r: r._should_auto_approve()).sudo().action_approve()
``` 
but will we have possible future inheritance errors?

Reply:
no, it would work normally